### PR TITLE
[FLINK-32226][client] Cleanup jobgraph file on submission failure

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java [hunk 1]
```diff
@@ -333,7 +359,7 @@ public CompletableFuture<JobID> submitJob(@Nonnull JobGraph jobGraph) {
                         () -> {
                             try {
                                 final java.nio.file.Path jobGraphFile =
-                                        Files.createTempFile("flink-jobgraph", ".bin");
+                                        Files.createTempFile(tempDir, "flink-jobgraph", ".bin");
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java:362]: I'm slightly concerned of this leaking a lot of strings in long-running applications that use the `RestClusterClient`.

[the remark this answers is not in the document]
Comment (rkhachatryan) [reply flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java:362]: Good point.

File: flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java [hunk 2]
```diff
@@ -434,16 +460,19 @@ public CompletableFuture<JobID> submitJob(@Nonnull JobGraph jobGraph) {
...
+                                jobGraphFile -> {
+                                    try {
+                                        Files.delete(jobGraphFile);
+                                    } catch (IOException e) {
+                                        LOG.warn(
+                                                "Could not delete temporary file {}.",
+                                                jobGraphFile,
+                                                e);
+                                    }
+                                }));
```
Comment (rkhachatryan) [inline flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java:475]: nit:
this seems a bit more readable to me:
```
        submissionFuture
                .exceptionally(ign -> null)   // ignore errors
                .thenAccept(
                        jobGraphFile -> {
...
```

**Review by rkhachatryan — APPROVED** [review]

File: flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java (hunk 2 above)
Comment (zentol) [reply flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java]: `exceptionally` simplifies the change quite a bit, added.

**Pull request merged**